zilliqa/src/consensus.rs

```diff
@@ -1471,7 +1501,19 @@ impl Consensus {
 
                 let current_head = self.head_block();
 
-                self.proposal(Proposal::from_parts(block, transactions), true)?;
+                self.proposal(
+                    Proposal::from_parts_and_hashes(
```

Comment:
I would assume instead here you want to reconstruct the `Proposal` you had at the start of this call to `recieve_block`? Won't this drop any `opaque_transactions` that were sent?

Reply:
The `opaque_transactions` will still be present as hashes on `block.transactions` - the `transactions` local variable here will only contain the non-opaque ones, whereas the `block` always contains _all_ hashes that should go into the block.

This `from_parts_and_hashes()` should be the exact inverse of `into_parts()`. As I've mentioned though the parameter naming in `from_parts_and_hashes()` is mediocre so I'll rename that.